### PR TITLE
grad_X

### LINKED ISSUE #10: add a grad_X in a following PR for sensitivity
Should we add a grad_X in a following PR for sensitivity?

_Originally posted by @pierrenodet in https://github.com/Orange-OpenSource/mislabeled/pull/8#discussion_r1887506694_

### PR DESCRIPTION
**Pull request by pierrenodet**
Fixes #10

### CODE REVIEW

File: mislabeled/probe/_linear.py
Comment (tfjgeorge) [inline mislabeled/probe/_linear.py:1]: instead of computing a large 3d tensor then select only the submatrix corresponding to the observed y, would it be possible to swap the order of ops ?

**Review by tfjgeorge — CHANGES_REQUESTED** [review]
I think it should be clearly stated that this is the gradient with respect to the loss. E.g. in https://github.com/Orange-OpenSource/mislabeled/blob/eb6ed9c133f8962a2070f9fa1fd468c0ad63e35e/mislabeled/probe/_sensitivity.py#L89 VOSG stands for "softmax gradient".

**Comment by pierrenodet** [comment]
Indeed, I have no preferences if Sensitivity implements dl_dX or dy_dX[:, y_observed]; we can keep the old implementation to be faithful to the mislabeled paper or use this one to generalize the approach to regression (if it makes sense ?) and other loss functions? Your call :)

**Comment by tfjgeorge** [comment]
IMO it makes more sense to use dl_dX (as is currently implemented) so it would be enough to keep it this way and change detector names/documentation to reflect the current implementation

**Comment by pierrenodet** [comment]
Ok, I added FiniteDiff in front of VoX detectors that used the FiniteDiff approach, and I dropped the S for VoSG detectors. The original VoG is now called FiniteDiffVoL(ogits)G, and the loss-based detectors are FiniteDiffVoG and VoG (for the linear estimators).

**Comment by tfjgeorge** [comment]
Sorry for being picky but it would make more sense if VoG stood for Variance of logits gradients, since this is the name used in the paper.

VoLG would instead stand for Variance of Loss Gradients: a different name for a different method than what is described in the paper.
